packages/headless/src/api/search/search-api-client.ts

```diff
@@ -117,80 +121,89 @@ export class SearchAPIClient {
 
     this.searchAbortController = null;
 
-    if (isSuccessSearchResponse(platformResponse)) {
+    const body = await response.json();
+    const payload = {response, body};
+
+    if (isSuccessSearchResponse(body)) {
       const processedResponse = await this.options.postprocessSearchResponseMiddleware(
-        platformResponse
+        payload
       );
       return {
         success: processedResponse.body,
       };
     }
 
     return {
-      error: unwrapError(platformResponse),
+      error: unwrapError(payload),
     };
   }
 
   async facetSearch(req: FacetSearchRequest) {
-    const platformResponse = await PlatformClient.call<FacetSearchResponse>({
+    const response = await PlatformClient.call({
       ...baseSearchRequest(req, 'POST', 'application/json', '/facet'),
       requestParams: pickNonBaseParams(req),
       ...this.options,
     });
+
+    const body = await response.json();
+    const payload = {response, body};
+
     const processedResponse = await this.options.postprocessFacetSearchResponseMiddleware(
-      platformResponse
+      payload
     );
 
     return processedResponse.body;
   }
 
   async recommendations(req: RecommendationRequest) {
-    const platformResponse = await PlatformClient.call<Search>({
+    const response = await PlatformClient.call({
       ...baseSearchRequest(req, 'POST', 'application/json', ''),
       requestParams: pickNonBaseParams(req),
       ...this.options,
     });
 
-    if (isSuccessSearchResponse(platformResponse)) {
-      return {
-        success: platformResponse.body,
-      };
+    const body = await response.json();
+
+    if (isSuccessSearchResponse(body)) {
+      return {success: body};
     }
 
     return {
-      error: unwrapError(platformResponse),
+      error: unwrapError({response, body}),
     };
   }
 
   async html(req: HtmlRequest) {
-    const response = await PlatformClient.call<string>({
+    const response = await PlatformClient.call({
       ...baseSearchRequest(req, 'POST', 'application/json', '/html'),
       requestParams: pickNonBaseParams(req),
       ...this.options,
     });
 
-    if (isSuccessHtmlResponse(response)) {
-      return {success: response.body};
+    const body = await response.text();
```

Comment:
This line is the main reason for the refactor, making it possible to parse the response from the `html` endpoint differently from the other endpoints.